app/models/registration_repository.rb

```diff
@@ -61,7 +61,41 @@ def self.store_search_query(session, query)
     end
   end
 
-  # Gets stored search query params
+  # restore search query to render search screen again
+  def self.restore_search_query_options(session)
+    session = session.dup
```

Comment:
After the persistent registrant record is created, there is no more use 
for the session object, and in fact retaining it is incorrect.

ejs

amekelburg wrote:
> In app/models/registration_repository.rb 
> <https://github.com/TrustTheVote-Project/Voter-Services-Portal/pull/12#discussion_r74040011>:
> 
>> @@ -61,7 +61,41 @@ def self.store_search_query(session, query)
>>      end
>>    end
>>  
>> -  # Gets stored search query params
>> +  # restore search query to render search screen again
>> +  def self.restore_search_query_options(session)
>> +    session = session.dup
> 
> Is there a particular reason we don't want to call delete on the actual 
> session object? In the interest in preserving past functionality, maybe 
> keep all the session.delete calls on the original session object, but 
> set the values on a new object that gets returned?
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/TrustTheVote-Project/Voter-Services-Portal/pull/12/files/a7096fe978d33ac0e686df5230af5f9f153ae5f7#r74040011>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AAD1fbmhrnMmuERb_KeZNTaMwc1hbTpKks5qeGOogaJpZM4Jf8nB>.
>